pkg/acceptance/node_test.go

```diff
@@ -28,120 +28,14 @@ func TestDockerNodeJS(t *testing.T) {
 }
 
 const nodeJS = `
-const fs     = require('fs');
-const pg     = require('pg');
-const assert = require('assert');
+set -e
+cd /testdata/node
 
-const config = {
-  user: 'root',
-  ssl: {
-    cert: fs.readFileSync(process.env.PGSSLCERT),
-    key:  fs.readFileSync(process.env.PGSSLKEY),
-  }
-};
-
-const client = new pg.Client(config);
-
-// Assert that the given error is not null and matches the given regex.
-function isError(err, expected, pgCode) {
-  if (!err) return new Error('expected non-null error');
-  if (pgCode && err.code !== pgCode) return new Error('expected error to have code ' + pgCode + ', got ' + err.code);
-  if (!expected.test(err.toString())) return new Error('expected error message to match ' + expected.toString() + ', but was ' + err.toString());
-  return null;
-}
-
-function testSelect(client) {
-  return new Promise(resolve => {
-    client.query("SELECT 1 as first, 2+$1 as second, ARRAY['\"','',''] as third", [%v], function (err, results) {
-      if (err) throw err;
-
-      assert.deepEqual(results.rows, [{
-        first: 1,
-        second: 5,
-        third: ['"', '', '']
-      }]);
-
-      resolve();
-    });
-  });
-}
-
-function testSendInvalidUTF8(client) {
-  return new Promise(resolve => {
-    client.query({
-      text: 'SELECT $1::STRING',
-      values: [new Buffer([167])]
-    }, function (err, results) {
-      let validationError = isError(err, /invalid UTF-8 sequence/, '22021');
-      if (validationError) throw validationError;
-      resolve();
-    });
-  });
-}
-
-function testSendNotEnoughParams(client) {
-  return new Promise(resolve => {
-    client.query({
-      text: 'SELECT 3',
-      values: ['foo']
-    }, function (err, results) {
-      let validationError = isError(err, /expected 0 arguments, got 1/, '08P01');
-      if (validationError) throw validationError;
-      resolve();
-    });
-  });
-}
-
-function testInsertArray(client) {
-  return new Promise(resolve => {
-    client.query({
-      text: 'CREATE DATABASE d',
-    }, (err) => {
-      if (err) throw err;
-      client.query({
-        text: 'CREATE TABLE d.x (y FLOAT[])',
-      }, (err) => {
-        if (err) throw err;
-        client.query({
-          text: 'INSERT INTO d.x VALUES ($1)',
-          values: [[1, 2, 3]],
-        }, (err) => {
-          if (err) throw err;
-          resolve();
-        });
-      });
-    });
-  });
-}
-
-function runTests(client, ...tests) {
-  if (tests.length === 0) {
-    return Promise.resolve();
-  } else {
-    return tests[0](client).then(() => runTests(client, ...tests.slice(1)));
-  }
-}
-
-client.connect(function (err) {
-  if (err) throw err;
-
-  runTests(client,
-    testSendInvalidUTF8,
-    testSendNotEnoughParams,
-    testInsertArray,
-    testSelect
-  ).then(result => {
-    client.end(function (err) {
-      if (err) throw err;
-    });
-  }).catch(err => {
-    client.end()
-    throw err;
-  });
-});
+export SHOULD_FAIL=%v
+npm install && npm test
```

Comment:
Does this work offline?

Reply:
Yes, as long as it's been run before to install the dependencies once, it won't try to hit the network to pull them down again (not that I had realized this was a concern beforehand).

Reply:
That's fine, was just checking.

Reply:
Going to try to instead install these globally in the docker image instead, so this shouldn't be a concern at all.